src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -1,227 +1,280 @@
-        }
-
-        final Regs regs = request.getRegs();
-
-        final String gpp = regs != null ? regs.getGpp() : null;
-        if (gpp != null) {
-            requestBuilder.gpp(gpp);
-        }
-
-        final List<Integer> gppSid = regs != null ? regs.getGppSid() : null;
-        if (CollectionUtils.isNotEmpty(gppSid)) {
-            requestBuilder.gppSid(gppSid);
-        }
-
-        final ExtRegs extRegs = regs != null ? regs.getExt() : null;
-        final String usPrivacy = extRegs != null ? extRegs.getUsPrivacy() : null;
-        if (usPrivacy != null) {
-            requestBuilder.usPrivacy(usPrivacy);
-        }
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<Imp> imps = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
+        if (bidRequest.getSite() != null) {
+            for (Imp imp : bidRequest.getImp()) {
+                try {
+                    final ExtImpConsumable impExt = parseImpExt(imp);
 
-        final Integer gdpr = extRegs != null ? extRegs.getGdpr() : null;
-        final User user = request.getUser();
-        final ExtUser extUser = user != null ? user.getExt() : null;
-        final String gdprConsent = extUser != null ? extUser.getConsent() : null;
-        if (gdpr != null || gdprConsent != null) {
-            final ConsumableBidGdpr.ConsumableBidGdprBuilder bidGdprBuilder = ConsumableBidGdpr.builder();
-            if (gdpr != null) {
-                bidGdprBuilder.applies(gdpr != 0);
+                    if (impExt.getSiteId() != 0 && impExt.getNetworkId() != 0 && impExt.getUnitId() != 0) {
+                        imps.add(modifyImp(imp, impExt));
+                    }
+                } catch (PreBidException e) {
+                    errors.add(BidderError.badInput(e.getMessage()));
+                }
             }
-            if (gdprConsent != null) {
-                bidGdprBuilder.consent(gdprConsent).build();
+            httpRequests.add(BidderUtil.defaultRequest(modifyBidRequest(bidRequest, imps), resolveHeaders(),
+                    this.endpointUrl + SITE_URI_PATH, mapper));
+        } else if (bidRequest.getApp() != null) {
+            for (Imp imp : bidRequest.getImp()) {
+                try {
+                    final ExtImpConsumable impExt = parseImpExt(imp);
+
+                    if (!Strings.isNullOrEmpty(impExt.getPlacementId())) {
+                        httpRequests.add(BidderUtil.defaultRequest(modifyBidRequest(bidRequest,
+                                Collections.singletonList(modifyImp(imp, impExt))), resolveHeaders(),
+                                this.endpointUrl + APP_URI_PATH + impExt.getPlacementId(), mapper));
+                    }
+                } catch (PreBidException e) {
+                    errors.add(BidderError.badInput(e.getMessage()));
+                }
```

Comment:
I suggest merge the App and Site cases into one since the only difference between them is the URL construction, so I believe it'll make the code neater

Reply:
After looking at the code more deeply there are more that is happening than just the url construction.   The app case extracts the placementId from the Imp and makes an httprequest for each imp instead of combining them as this is how our system works in the backend.  Combining these makes the code way more messy.